src/modules/rolle/domain/rolle.ts

```diff
@@ -91,13 +92,19 @@ export class Rolle<WasPersisted extends boolean> {
             istTechnisch,
             serviceProviderData ?? [],
         );
-        //Replace service providers with new ones.
-        for (const serviceProviderId of serviceProviderIds) {
-            const result: void | DomainError = await rolleToUpdate.attachServiceProvider(serviceProviderId);
+        //Replace service providers with new ones
+        const attachmentResults: (void | DomainError)[] = await Promise.all(
+            serviceProviderIds.map(async (serviceProviderId: string) => {
+                return rolleToUpdate.attachServiceProvider(serviceProviderId);
+            }),
+        );
+
+        for (const result of attachmentResults) {
```

Comment:
You do not need a for loop here, just check if there is any result is a DomainError